server/neptune/workflows/internal/github/commit.go

```diff
@@ -1,6 +1,18 @@
 package github
 
+import "fmt"
+
 type Commit struct {
-	Ref    string
+	Ref    Ref
 	Author User
 }
+
+type Ref struct {
+	Name string
+	Type string
+}
+
+func (r Ref) String() string {
+	// We only support branch type refs
+	return fmt.Sprintf("refs/heads/%s", r.Name)
```

Comment:
feels easy enough to just do:
```go
if type == "branch"
    return fmt.Sprintf("refs/heads/%s", r.Name)
  else if type == "tag"
    return fmt.Sprintf("refs/tags/%s", r.Name)
```

if for some reason we ever supported it, we'd have to track this down.  Either fail or support it, but we should be explicit.